### PR TITLE
Update KeyInfo and KeyId

### LINKED ISSUE #248: compare keys by fingerprints whenever possible. Store key fingerprints in db
We should start saving fingerprints in the storage (in addition to longids), and compare by fingerprints wherever possible instead of longids. It's safer - collisions less likely. I'll make another issue for this.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/pull/245#discussion_r458311381_

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR updates way of storing KeyInfo and KeyId

issue #248
----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (explain why)
--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any) - **only partially**
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Controllers/Settings/KeySettings/Key Detail Info/KeyDetailInfoViewDecorator.swift [hunk 1]
```diff
@@ -36,7 +25,8 @@ struct KeyDetailInfoViewDecorator: KeyDetailInfoViewDecoratorType {
         case .keyWord:
             title = "key_settings_detail_key_words".localized + ":" + " "
             return title.attributed(.medium(16))
-                + keyId.keywords.attributed(.regular(16), color: .main)
+            // TODO: - Tom?
```
Comment (Kharchevskyi) [inline FlowCrypt/Controllers/Settings/KeySettings/Key Detail Info/KeyDetailInfoViewDecorator.swift:28]: @tomholub previously we had keywords in KeyDetailInfoViewCointroller. 
Should keywords be part of KeyDetail? Or this part should be removed?

**Comment by Kharchevskyi** [comment]
@tomholub this task looks a bit complicated(
Is it possible to keep PRs as small as possible and create next issue to finish this functionality?

**Comment by tomholub** [comment]
> @tomholub this task looks a bit complicated(
> Is it possible to keep PRs as small as possible and create next issue to finish this functionality?

Sure, as long as the intermediate steps don't break the app

File: FlowCrypt/Controllers/Settings/KeySettings/Key Detail Info/KeyDetailInfoViewDecorator.swift (hunk 1 above)
Comment (tomholub) [reply FlowCrypt/Controllers/Settings/KeySettings/Key Detail Info/KeyDetailInfoViewDecorator.swift:28]: please remove keywords from all parts of the app. They are not useful, I'm deprecating them

File: FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift [hunk 2]
```diff
@@ -110,7 +110,7 @@ extension EncryptedStorage: LogOutHandler {
             let keys = storage.objects(KeyInfo.self)
                 .filter { $0.account == email }
             let passPhrases = storage.objects(PassPhraseObject.self)
-                .filter { keys.map(\.longid).contains($0.longid) }
+                .filter { keys.map(\.primaryLongid).contains($0.longid) }
```
Comment (tomholub) [inline FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift:113]: Later this should be compared by `primaryFingerprint` vs `$0.fingerprint` - it's ok to do it in a followup PR

File: FlowCrypt/Functionality/Services/Key Services/KeyService.swift [hunk 3]
```diff
@@ -69,7 +69,7 @@ final class KeyService: KeyServiceType {
         // get all private keys with already saved pass phrases
         var privateKeys = keysInfo
             .compactMap { keyInfo -> PrvKeyInfo? in
-                guard let passPhrase = storedPassPhrases.first(where: { $0.longid == keyInfo.longid }) else {
+                guard let passPhrase = storedPassPhrases.first(where: { $0.longid == keyInfo.primaryLongid }) else {
```
Comment (tomholub) [inline FlowCrypt/Functionality/Services/Key Services/KeyService.swift:72]: Same here

File: FlowCrypt/Models/Realm Models/KeyInfo.swift [hunk 4]
```diff
@@ -13,12 +13,24 @@ enum KeySource: String {
 enum KeyInfoError: Error {
     case missedPrivateKey(String)
     case notEncrypted(String)
+    case missedKeyIds
 }
 
 final class KeyInfo: Object {
+    var primaryFingerprint: String? {
+        allFingerprints.first
+    }
```
Comment (tomholub) [inline FlowCrypt/Models/Realm Models/KeyInfo.swift:22]: A KeyInfo object always must have a primaryFingerprint, so I would force this, or check for it and throw.

File: FlowCrypt/Models/Realm Models/KeyInfo.swift [hunk 5]
```diff
@@ -33,9 +45,15 @@ final class KeyInfo: Object {
             assertionFailure("Will not store Private Key that is not fully encrypted") // crash tests
             throw KeyInfoError.notEncrypted("Will not store Private Key that is not fully encrypted")
         }
+        guard keyDetails.ids.isNotEmpty else {
+            assertionFailure("KeyDetails KeyIds should not be empty")
+            throw KeyInfoError.missedKeyIds
```
Comment (tomholub) [inline FlowCrypt/Models/Realm Models/KeyInfo.swift:50]: The right spelling here is `missing` rather then `missed`

**Review by tomholub — COMMENTED** [review]
Looks good - if you want, you can clean up the comments below and then do a followup PR to continue the work.

File: FlowCrypt/Models/Realm Models/KeyInfo.swift (hunk 4 above)
Comment (Kharchevskyi) [reply FlowCrypt/Models/Realm Models/KeyInfo.swift:22]: actually this is already checked here 
```
guard keyDetails.ids.isNotEmpty else {
  assertionFailure("KeyDetails KeyIds should not be empty")
  throw KeyInfoError.missedKeyIds
}

...

self.allFingerprints.append(objectsIn: keyDetails.ids.map(\.fingerprint))
```

**Review by tomholub — APPROVED** [review]

**Pull request merged**